ui/turn-order-spell-api.js

```diff
@@ -0,0 +1,56 @@
+on('ready', () => {
+  const getTurnOrder = () => (
+    Campaign().get('turnorder') === '' ?
+      [] :
+      JSON.parse(Campaign().get('turnorder'))
+  );
+  const setTurnOrder = (turnOrder) => Campaign().set({ turnorder: JSON.stringify(turnOrder) });
+
+  function addSpellDurationToTurnOrder(event, commandValues) {
+    const { who } = event;
+    const [characterId, spellName, spellDuration] = commandValues;
+    const pageid = Campaign().get('playerpageid');
+    const tokens = findObjs({
+      _pageid: pageid,
+      _type: 'graphic'
+    });
+    const matchingCharacterToken = tokens.find((token) => token.get('represents') === characterId);
+
+    // const player = getObj('player', characterId);
+    if (!matchingCharacterToken) {
+      log(`Unable to get player token for ${characterId}`);
+    }
+    const playerName = matchingCharacterToken ? matchingCharacterToken.name : '';
+    const turnOrder = getTurnOrder();
+    const customEntry = `${playerName} ${spellName}`;
+    log(`Adding ${spellName} to turn order`);
+    const entry = {
+      id: '-1',
+      source: 'addSpellDurationToTurnOrder',
+      custom: customEntry,
+      pr: parseInt(spellDuration) || 0,
+      formula: -1,
+    };
+    setTurnOrder([entry, ...turnOrder]);
```

Comment:
I am placing the new entry ahead of the current user. This means that it will decrement right before we get to the current user.